components/script/dom/location.rs

```diff
@@ -217,7 +230,11 @@ impl LocationMethods for Location {
     // https://html.spec.whatwg.org/multipage/#dom-location-protocol
     fn SetProtocol(&self, value: USVString) -> ErrorResult {
         self.check_same_origin_domain()?;
-        self.set_url_component(value, UrlHelper::SetProtocol);
+        // If copyURL's scheme is not an HTTP(S) scheme, then terminate these steps.
+        let scheme = value.split(':').next().unwrap();
+        if scheme.eq_ignore_ascii_case("http") || scheme.eq_ignore_ascii_case("https") {
```

Comment:
The [spec](https://html.spec.whatwg.org/multipage/history.html#dom-location-protocol) contains this note:

> Because the URL parser ignores multiple consecutive colons, providing a value of "https:" (or even "https::::") is the same as providing a value of "https".

Does this allow for multiple consecutive colons?

If this is currently not tested in the existing tests, you could also consider adding such a testcase.

Reply:
Ok, I'll write some tests!